### DISCUSSION TITLE
How to get the element which handles a bubbled event?

### DISCUSSION THREAD

**Discussion by northern-mindset** [alpinejs/alpine#1373] [1. Help]
In case the event was originated (`e.target`) on one of the children of an element which actually handles the event (`'x-on'), how can I get the latter one inside the handler? `this` is bound to the component data…

**Comment by ryangjchandler** [comment]
If you're using a function, you will receive an `$event` object as the first argument. Otherwise, you can use `$event` inline:

```html
<button x-on:click="$event.target">
```

```html
// or using an event function
<button x-on:click="handleClick">

<script>
function data() {
    return {
        handleClick($event) {}
    }
}
</script>
```

**Reply by northern-mindset** [reply]
The `$event.target` refers to the element to which the event was originally dispatched. However my handler is attached to the outer element, thus I'm getting the event in the bubbling phase. And while inspecting the `$event.target` I don't see any reference to my outer element in it.

Sample DOM:

```
<button x-on:click="console.log($event.target)" >
	  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#4b5563">
	  ...
	  </svg>
</button>
```

In this case `svg` is the target, or even `path` inside `svg`.

The idea is that I want to have a generalized handler that would obtain a parameter from the element's data-attribute. And I don't want to embed the parameter into the alpine directive like `x-on:click=handler(parameter)`.

Is there any way of getting the reference to the element which captured the event inside the `x-on` handler?

**Reply by SimoTod** [reply]
Have you tried $event.currentTarget? https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget

**Reply by northern-mindset** [reply]
Before asking the question I had inspected the event in the console output, and noticed the `currentTarget` property. The name was promising, but the value was `null`, therefore I disregarded this property.

But now I have attentively read the documentation and found a note:

> The value of `event.currentTarget` is only available while the event is being handled. If you `console.log()` the event object, storing it in a variable, and then look for the `currentTarget` key in the console, its value will be `null`. Instead, you can either directly `console.log(event.currentTarget)` to be able to view it in the console…

Therefore it works, but the behavior of this property is still mysterious to me. Anyhow, thank you @SimoTod.
